src/array.cr

```diff
@@ -1381,6 +1381,19 @@ class Array(T)
     self
   end
 
+  # Replaces the contents of `self` with the contents of *other*.
+  # This resizes the Array to a greater capacity but does not free memory if the given array is smaller.
+  #
+  # ```
+  # a1 = [1, 2, 3]
+  # a1.replace([1])
+  # a1                    # => [1]
+  # a1.remaining_capacity # => 3
+  # a1                    # => [1]
```

Comment:
This is redundant (and its obvious that `remaining_capacity` doesn't alter the array)

```suggestion
```